### ISSUE TITLE
Add accessible labels to component pages in the guidance site to aid screen reader navigation

### ISSUE THREAD

**Issue by DBD324** [mi6/ic-design-system#609]
## Summary

From a11y testing it was observed that some component pages can be mistaken for other components. 

For example, the left hand nav uses a series of links that announce 'Link, page name'. In examples where the page name exactly matches the SR announcement for that component this can be confusing when navigating with a SR. The button page is announced 'Link, button' for example. This could indicate that the element is actually a button, and not a link to the button page. 

We can use accessible labels to add more context to the page links so that they are more easily understood. For example, 'Link, Button component' or 'Link, Button guidance'. We need to ensure that the visible label is included within the accessible label to meet criteria 'Label in name'. 

## 💬 Description

Review each component page and compare the announced element ('link, button') with the announcement for when focussing on an instance of that component. If they match or are too similar, then consider updating the accessible label to add more context. 

A suggestion is to use the following structure for each page's accessible label:
'[Component name] component'

This would result in an SR announcement 'Link, Button component' or 'Link, link component'. 

For consistency, we should append 'Component' to every component page's accessible label. 

## 💰 Use value

Makes it easier for users of assistive technology to navigate through the website and understand which pages they navigate to. 

## 📝 Acceptance Criteria

If relevant, describe in full detail the different interactions and edge cases that the component or patterns needs to fulfil.

_Given_ that I am navigating through the components section left hand nav
_When_ focussing on each link element
_Then_ the link should be announced in the following format:
- 'Link, [page name] component'

## ✏ Designs

If there's a Figma design file (or other mock-up), include it here.

## 🧾 Guidance
If there's written guidance or documentation, include a link to it here.

## Additional info

Tell us anything else useful to help us understand your suggestion.
